### PR TITLE
add db,swagger

### PR DESCRIPTION
**Pull request by MustafaMadkour**
# Navigation Links API
## _Description_
 API to create navigation bar links and retrieve them from a mongodb database.
Each of these links has a title, description, links-array and images array.  
The API has 4 routes to get a single document, all documents, creating a document and a Sawagger UI route.

### Stack:
We used Nestjs - in typescript -and Mongodb as a databases.

### Requests:  
we have 3 requests available:
- GET  `localhost:3000/navs` to get all navigation links.
- POST `localhost:3000/navs` to create a new navigation link
- GET `localhost:3000/navs/:id` to get a navigation link by its id

### Tasks to complete:
- [x] db
- [x] Swagger document
- [ ] use typeORM

### Links:
`localhost:3000/navs` > to get all documents in db
`localhost:3000/navs/:id` > to get a single link with its id.
`localhost:3000/api` > to get swagger document.

### DB document interface:
- `_id` : generated by mongodb
- `title`: a string title for the link.
- `description`: a string for descripping this navigation link
- `links`: an array of strings to contain all links to include in this navigation section
- `images`: an array of strings to contain all images to be included in this navigation section.

### CODE REVIEW

File: api/package.json [hunk 1]
```diff
@@ -23,10 +23,14 @@
   "dependencies": {
     "@nestjs/common": "^7.0.0",
     "@nestjs/core": "^7.0.0",
+    "@nestjs/mongoose": "^7.0.2",
```
Comment (Nilegfx) [inline api/package.json:26]: use typeORM

File: api/src/config/keys.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export default {
+  mongoURI: 'mongodb+srv://mustafa:mmmkkkmmm@cluster0.zj3wa.mongodb.net/prsresume?retryWrites=true&w=majority'
```
Comment (Nilegfx) [inline api/src/config/keys.ts:2]: no, he didn't do this intentionally.. he didn't do this intentionally.. he didn't do this intentionally.. he didn't do this intentionally.. he didn't do this intentionally.. 

NOOOOOOOOOOOOOO, He did!! 🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️

**Review by Nilegfx — CHANGES_REQUESTED** [review]

File: api/src/config/keys.ts (hunk 2 above)
Comment (AbdoMgdy) [reply api/src/config/keys.ts:2]: oh newbies .... what can you do !! :laughing:

Comment (Nilegfx) [reply api/src/config/keys.ts:2]: maho el fekra eny 3aref en mustafa mesh newbie 😂

Comment (AbdoMgdy) [reply api/src/config/keys.ts:2]: howa kan mest3gel bs w 2ableto mshakel kteer m3 git

Comment (amagdy46) [inline api/src/config/keys.ts:2]: replace the this with an environment variable for security concerns

Comment (MustafaMadkour) [reply api/src/config/keys.ts:2]: ده توتر مصحوب بالمزيد من الغباء 
لا فعلا احزننك جدا ... والله عيب فعلا

File: api/package.json (hunk 1 above)
Comment (MustafaMadkour) [reply api/package.json:26]: I still need more time to get very familiar with nestjs best practices.
rule #1 : it's working right now and we can refactor it.

**Comment by amagdy46** [comment]
I like how you wrote the description well structured and clean. Good Job !

**Pull request closed without merging**